### PR TITLE
Add cats-effect to the community build

### PR DESCRIPTION
**Pull request by vasilmkd**
Cats Effect 2 added under "cats-effect-2".
Cats Effect 3 added under "cats-effect-3".

cc @djspiewak

### CODE REVIEW

File: community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala [hunk 1]
```diff
@@ -300,6 +300,16 @@ object projects:
...
   )
 
+  lazy val catsEffect2 = SbtCommunityProject(
+    project        = "cats-effect-2",
+    sbtTestCommand = "test"
+  )
+
+  lazy val catsEffect3 = SbtCommunityProject(
+    project        = "cats-effect-3",
+    sbtTestCommand = "test"
```
Comment (djspiewak) [inline community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala:310]: Oh, this one probably needs to be `testIfRelevant` for the moment. I'm working on fixing that.

**Comment by vasilmkd** [comment]
The cats effect build failed with the following exception:

`sbt.librarymanagement.ResolveException: Error downloading ch.epfl.lamp:dotty-library_sjs1_0.28:0.28.0-bin-SNAPSHOT`

**Comment by smarter** [comment]
I think `` (publishLocal in `dotty-library-bootstrappedJS`).value `` needs to be added to `prepareCommunityBuild` in Build.scala: https://github.com/lampepfl/dotty/blob/c104e9b1b7c028f114f8f3c0331198a9587391b5/project/Build.scala#L1253

**Comment by vasilmkd** [comment]
That did the trick. Thank you @smarter.

**Comment by anatoliykmetyuk** [comment]
Thanks for contributing this @vasilmkd! All of the Dotty community projects live in forks under dotty-staging organization – so that we have the capability to fix them fast if they break. I've created such a fork for cats-effects: https://github.com/dotty-staging/cats-effect – could you point the submodule to this fork instead?

**Comment by vasilmkd** [comment]
@anatoliykmetyuk On it right now.

**Review by anatoliykmetyuk — APPROVED** [review]

**Comment by anatoliykmetyuk** [comment]
Looks like this project fails the CI once in a while:

https://github.com/lampepfl/dotty/runs/1131886366?check_suite_focus=true
https://github.com/lampepfl/dotty/runs/1131886366?check_suite_focus=true

@vasilmkd @djspiewak do you have an idea why this is happening?

**Comment by vasilmkd** [comment]
The tests that fail are really hard to properly write and we know them to be slightly non-deterministic and therefore flaky. May I asked what environment these tests run in?

**Comment by anatoliykmetyuk** [comment]
The CI definition of the community build step is here: https://github.com/lampepfl/dotty/blob/04659bec64d98cc483c9288f91b04a5b274989bb/.github/workflows/ci.yaml#L99
And the Docker container used is defined as follows: https://github.com/lampepfl/dotty-ci/blob/master/dotty-docker/Dockerfile

**Comment by anatoliykmetyuk** [comment]
Is it possible to disable these tests in the dotty-staging fork so that the entire CI doesn't fail non-deterministically?

**Comment by vasilmkd** [comment]
I will push a commit to do that in a bit, is that ok?

**Comment by vasilmkd** [comment]
I opened a PR here.
https://github.com/dotty-staging/cats-effect/pull/1

**Comment by vasilmkd** [comment]
@anatoliykmetyuk @smarter What does the process for keeping the `dotty-staging/cats-effect` fork up to date with upstream? For example, we have a fix for the flaky tests pushed to `typelevel/cats-effect` and we'd like to incorporate them into the community build. How do we do that? Is it maybe possible to restore the git histories to be identical, since they have diverged with the commit that disabled the flaky tests. Thank you.

cc @djspiewak

**Comment by smarter** [comment]
> Is it maybe possible to restore the git histories to be identical, since they have diverged with the commit that disabled the flaky tests.

Yes, but we have to keep the old history in a branch in the fork, so that each dotty commit still points to reachable commits in the submodules.

**Comment by vasilmkd** [comment]
This might be a stupid question, I just want to be sure I understood properly. If I want to restore the `series/2.x` (the main branch) to its current state as it is in `typelevel/cats-effect`, I should first branch off the current `series/2.x` branch in `dotty-staging/cats-effect` in order to save the history and force push the main branch?

**Comment by smarter** [comment]
Yes, and push the saved branch too :) I suggest putitng the date in the name of the branch.

**Comment by vasilmkd** [comment]
@smarter Does `series/2.x-20.09.2020` work?

**Comment by smarter** [comment]
There's only one true date format: https://xkcd.com/1179/ :)

**Comment by vasilmkd** [comment]
Agreed.

**Comment by vasilmkd** [comment]
https://github.com/dotty-staging/cats-effect/tree/series/2.x-2020-09-20

**Comment by vasilmkd** [comment]
Thank you @smarter for the quick replies and the help.

**Pull request merged**